partitioned-heat-conduction/solver-fenics/heat_pySDC.py

```diff
@@ -0,0 +1,346 @@
+For more information on pySDC, see also "https://parallel-in-time.org/pySDC/"
+'''
+
+# initialize level parameters
+level_params = dict()
+level_params['restol'] = 1e-11
+level_params['dt'] = pySDC_dt
+
+# initialize step parameters
+step_params = dict()
+step_params['maxiter'] = 40
+
+# initialize sweeper parameters
+sweeper_params = dict()
+sweeper_params['quad_type'] = 'LOBATTO'
+sweeper_params['num_nodes'] = 4
+
+# initialize problem parameters
+problem_params = dict()
+problem_params['mesh'] = domain_mesh
+problem_params['function_space'] = V
+problem_params['coupling_boundary'] = coupling_boundary
+problem_params['remaining_boundary'] = remaining_boundary
+problem_params['solution_expr'] = u_D
+problem_params['forcing_term_expr'] = forcing_expr
+problem_params['precice_ref'] = precice
+problem_params['coupling_expr'] = coupling_expression
+problem_params['participant_name'] = participant_name
+
+# initialize controller parameters
+controller_params = dict()
+controller_params['logger_level'] = 30
+
+# fill description dictionary for easy step instantiation
+description = dict()
+description['problem_class'] = fenics_heat_2d
+description['problem_params'] = problem_params
+description['sweeper_class'] = imex_1st_order_mass
+description['sweeper_params'] = sweeper_params
+description['level_params'] = level_params
+description['step_params'] = step_params
+
+# Controller for time stepping
+controller = controller_nonMPI(num_procs=1, controller_params=controller_params, description=description)
+
+# Reference to problem class for easy access to exact solution
+P = controller.MS[0].levels[0].prob
+
+##################################################################
+
+    
+
+#################################################################
+# Coupling loop
+#################################################################
+
+while precice.is_coupling_ongoing():
+
+    # write checkpoint
+    if precice.requires_writing_checkpoint():
+        precice.store_checkpoint(u_n, t, n)
+
+        # output solution and reference solution at t_n+1 and substeps (read from buffer)
+        print('output u^%d and u_ref^%d' % (n, n))
+        for sample in u_write:
+            temperature_out << sample
+
+        for sample in ref_write:
+            ref_out << sample
+
+        for sample in error_write:
+            error_out << error_pointwise
+
+    precice_dt = precice.get_max_time_step_size()
+    dt.assign(np.min([pySDC_dt, precice_dt]))
+
+    # Update start time of the current preCICE time step within
+    # the problem class.
+    # This is necessary, that the coupling expression update in the problem class
+    # is executed correctly.
+    P.set_t_start(t)
+
+    # Retrieve the result at the end of the timestep from pySDC
+    # Possible statistics generated by pySDC can be accessed via the 'stats' variable if needed
+    uend, stats = controller.run(u_n, t0=t, Tend=t + float(dt))
```

Comment:
```suggestion
    uend, _ = controller.run(u_n, t0=t, Tend=t + float(dt))
```
It's convention in python to replace unused variables with a `_`. You do not use `stats`, do you?

Even better would be `uend = controller.run(...)` but I guess the function always returns two values. In this case ignoring the `stats` and only keeping the `uend` is cleaner.

Reply:
Fixed. e889225ca17bee24f04072239004552831f7068d.